app/assets/javascripts/Components/Result/url_viewer.jsx

```diff
@@ -3,19 +3,85 @@ import React from "react";
 export class URLViewer extends React.Component {
   constructor(props) {
     super(props);
-    //TODO: Remove state url and replace its usage with url props when backend is working
     this.state = {
-      url: "https://www.youtube.com/embed/dQw4w9WgXcQ", // Sample Youtube embed link
-      //url: "https://drive.google.com/file/d/1yQh8bzo4IzmtSt-qZ0Vb-BzzEXsp45v_/preview" // sample google drive embed link
-      //url: "https://play.library.utoronto.ca/watch/4493e554354a7e51bdf90f3197185b5f" // sample mymedia link
+      url: "",
     };
   }
 
+  componentDidMount() {
+    this.configDisplay();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.externalUrl !== this.props.externalUrl) {
+      this.configDisplay();
+    }
+  }
+
+  configDisplay = () => {
+    try {
+      const url = new URL(this.props.externalUrl);
+      const youtube_view_is_set = this.configureOEmbedPreview(
+        url.toString(),
+        "https://www.youtube.com/oembed"
+      );
+      if (!youtube_view_is_set) {
+        switch (url.hostname) {
+          case "docs.google.com":
+          case "drive.google.com":
+            this.configureGoogleDrivePreview(url);
+            break;
+          default:
+            this.setDefaultState();
+        }
+      }
+    } catch (e) {
+      this.setDefaultState();
+    }
+  };
+
+  setDefaultState = () => {
+    this.setState({
+      url: "",
+    });
+  };
+
+  configureGoogleDrivePreview = url => {
+    const path = url.pathname.split("/");
+    if (path[1] === "forms") {
+      url.pathname = url.pathname.replace(/(\/[^\/]+)$/, "/viewform");
+    } else {
+      url.pathname = url.pathname.replace(/(\/[^\/]+)$/, "/preview");
+    }
+    this.setState({
+      url: url.toString(),
+    });
+  };
+
+  configureOEmbedPreview = (url, oembedUrl) => {
+    $.get(oembedUrl, {format: "json", url: url}).then(res => {
+      const match = res.html.match(/src="(\S+)"/);
+      if (match.length === 2) {
+        this.setState({
+          url: match[1],
+        });
+        return true;
+      }
+    });
+    return false;
+  };
+
   render() {
```

Comment:
Also can we please also render the original url somewhere and make it clickable:

```html
<a href={this.state.url}>{this.state.url}</a>
```

or this.props.externalUrl depending on which part of the if statement you're in

Reply:
This will be added in a separate PR